spec/javascripts/govuk_publishing_components/lib/auto-scroll-tracker-spec.js

```diff
@@ -0,0 +1,234 @@
+/* eslint-env jasmine, jquery */
+var GOVUK = window.GOVUK || {}
+
+describe('GOVUK.AutoScrollTracker', function () {
+  var scrollTracker, scrollTracker2
+
+  beforeEach(function () {
+    window.GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+    jasmine.clock().install()
+    spyOn(GOVUK.analytics, 'trackEvent')
+  })
+
+  afterEach(function () {
+    stopComponent(scrollTracker)
+    if (scrollTracker2) {
+      stopComponent(scrollTracker2)
+    }
+    jasmine.clock().uninstall()
+
+    window.GOVUK.deleteCookie('cookies_policy')
+    window.GOVUK.analyticsVars.scrollTrackerStarted = false
+
+    if (GOVUK.analytics.trackEvent.calls) {
+      GOVUK.analytics.trackEvent.calls.reset()
+    }
+  })
+
+  // remove all the event listeners, otherwise conflicts occur
+  function stopComponent (tracker) {
+    window.removeEventListener('scroll', tracker.scrollEvent)
+    window.removeEventListener('resize', tracker.resizeEvent)
+    clearInterval(tracker.interval)
+  }
+
+  it('should only initialise once on a page', function () {
+    var el = document.createElement('div')
+    scrollTracker = new GOVUK.Modules.AutoScrollTracker(el)
+    spyOn(scrollTracker, 'getWindowDetails')
+    scrollTracker.init()
+
+    var el2 = document.createElement('div')
+    scrollTracker2 = new GOVUK.Modules.AutoScrollTracker(el2)
+    spyOn(scrollTracker2, 'getWindowDetails')
+    scrollTracker2.init()
+
+    expect(scrollTracker.getWindowDetails).toHaveBeenCalled()
+    expect(scrollTracker2.getWindowDetails).not.toHaveBeenCalled()
+  })
+
+  describe('when tracking headings', function () {
+    var el
+
+    beforeEach(function () {
+      var h3pos = window.innerHeight + 1000
+      var FIXTURE =
+        '<div style="position: absolute; top: 0px;">' +
+          '<h1>Heading 1</h1>' +
+          '<div id="wrapper">' +
+            '<h2 style="display: none;">Heading 2</h2>' +
+            '<h3 style="margin-top: ' + h3pos + 'px;">Heading 3</h3>' +
+          '</div>' +
+        '</div>'
+      el = document.createElement('div')
+      el.setAttribute('data-track-type', 'headings')
+      el.innerHTML = FIXTURE
+      document.body.appendChild(el)
+      scrollTracker = new GOVUK.Modules.AutoScrollTracker(el)
+      scrollTracker.init()
+    })
+
+    afterEach(function () {
+      document.body.removeChild(el)
+    })
+
+    it('should send a tracking event on page load for headings that are already visible', function () {
```

Comment:
```suggestion
    it('should send a tracking event on initialisation for headings that are already visible', function () {
```

Since it doesn't necessarily need to be page load.

Reply:
Changed, thanks.